modules/core/src/opencl/runtime/opencl_core.cpp

```diff
@@ -177,6 +177,54 @@ static void *GetHandle(const char *file)
     return handle;
 }
 
+#ifdef __ANDROID__
+
+static const char *defaultAndroidPaths[] = {
+    "/system/lib64/libOpenCL.so",
```

Comment:
Please keep the default "libOpenCL.so" too.

Also `getRuntimePath(defaultAndroidPaths[i])` call below should be used with `"libOpenCL.so"` only.
(otherwise it would attempt to load multiple times the same library)

Reply:
Will `LD_LIBRARY_PATH` work with these approach?

Reply:
> Will LD_LIBRARY_PATH work with this approach?

@dkurt we are making our own call directly to `dlopen` instead of using the system's dynamic link loader so `LD_LIBRARY_PATH` is not relevant here.

Reply:
But it should be considered if none of the folders contain OpenCL library.

Reply:
https://www.man7.org/linux/man-pages/man3/dlopen.3.html

`dlopen()` is affected by `LD_LIBRARY_PATH`.

But using of `LD_LIBRARY_PATH` is not trivial on Android